### PR TITLE
Add activations as a property of layers

### PR DESCRIPTION
**Pull request by ludwigschubert**

### CODE REVIEW

**Comment by colah** [comment]
This looks great -- super excited to have activations easily accessible! This also feels like a nice step towards layers being an external data structure. :)

My main question is whether we should make more forward looking changes before merging:

* Not sure if this is worth doing now, but I'm tempted to rename the "conv" tag to "spatial". I just didn't think very hard when I originally wrote that.
* Some more possible tags:
  * "dense" vs "spatial"
  * "pool" "concat" "residual" "softmax"
  * more specific "MaxPool", "Relu", etc.
* we may want to allow pointers. eg:
  * "concat" layers might have a pointer "children": [conatonated layers]
  * "softmax" layers might have a pointer "logits"
* In the future, it might be nice to have a "downsample_factor", describing ratio of spatial dimensions in input to that layer (although, obviously, it wouldn't be exact due to padding shenanigans)
